torchaudio/csrc/CMakeLists.txt

```diff
@@ -119,31 +137,12 @@ if (BUILD_TORCHAUDIO_PYTHON_EXTENSION)
     set_target_properties(_torchaudio PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
   endif()
 
-  if (BUILD_SOX)
```

Comment:
I don't think this really affects the end result, but just curious why several components originally under the `_torchaudio.so` section are being moved to the `libtorchaudio` build

Reply:
Now `libtorchaudio` contains the implementation related to those definitions, such as (CUDA, Kaldi, libsox except file-like object I/O) meanwhile `_libtorchaudio` only contains libsox file-like object I/O. So these compiler definitions (`USE_CUDA`, `INCLUDE_KALDI` and `INCLUDE_SOX`) need to be defined in `libtorchaudio`. (and propagated to `_torchaudio` via `PUBLIC` keyword so that we only need to set them in one place).

Now that I think of it, probably, it's better to prefix these compiler definitions with `TORCHAUDIO_`.